src/overlays/actors/ovl_En_Ma4/z_en_ma4.c

```diff
@@ -21,78 +94,1011 @@ const ActorInit En_Ma4_InitVars = {
+                break;
+        }
+    }
+}
+
+void EnMa4_SetupDialogueHandler(EnMa4* this) {
+    if (this->type != MA4_TYPE_ALIENS_WON) {
+        EnMa4_ChangeAnim(this, 1);
+    } else {
+        EnMa4_ChangeAnim(this, 15);
+    }
+    this->actionFunc = EnMa4_DialogueHandler;
+}
+
+void EnMa4_DialogueHandler(EnMa4* this, GlobalContext* globalCtx) {
+    s32 temp_v0;
+
+    switch (func_80152498(&globalCtx->msgCtx)) {
+        default:
+            break;
+
+        case 4: // Player answered a question
+            EnMa4_HandlePlayerChoice(this, globalCtx);
+            break;
+
+        case 5: // End message block
+            EnMa4_ChooseNextDialogue(this, globalCtx);
+            break;
+
+        case 6: // End conversation
+            if (func_80147624(globalCtx) != 0) {
+                if ((globalCtx->msgCtx.unk_120B1 == 0) ||
+                    (((temp_v0 = gSaveContext.inventory.questItems) & gBitFlags[0x12]) == 0)) {
+                    EnMa4_SetupWait(this);
+                }
+            }
+
+        case 0:
+        case 1:
+        case 2:
+        case 3:
+            break;
+    }
+
+    if (this->type != MA4_TYPE_ALIENS_WON) {
+        Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 5, 2000, 1000);
+    }
+}
+
+void EnMa4_SetupBeginHorsebackGame(EnMa4* this) {
+    this->actionFunc = EnMa4_BeginHorsebackGame;
+}
+
+void EnMa4_BeginHorsebackGame(EnMa4* this, GlobalContext* globalCtx) {
+    globalCtx->nextEntranceIndex = 0x6400;
+    gSaveContext.unk_3F4A = 0xFFF0;
+    globalCtx->sceneLoadFlag = 0x14;
+    globalCtx->unk_1887F = 0x50;
+    gSaveContext.nextTransition = 3;
+}
+
+void EnMa4_HorsebackGameCheckPlayerInteractions(EnMa4* this, GlobalContext* globalCtx) {
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        // "You're feeling confident"
+        func_801518B0(globalCtx, 0x336E, &this->actor);
+        this->actionFunc = EnMa4_HorsebackGameTalking;
+    } else if ((gSaveContext.unk_3DE0[4] < 115 * 100)) { // timer < 115 seconds
+        func_800B8614(&this->actor, globalCtx, 100.0f);
+    }
+}
+
+void EnMa4_HorsebackGameTalking(EnMa4* this, GlobalContext* globalCtx) {
+    if (func_800B867C(&this->actor, globalCtx)) {
+        this->actionFunc = EnMa4_HorsebackGameWait;
+    }
+}
+
+void EnMa4_InitHorsebackGame(EnMa4* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    globalCtx->interfaceCtx.unk_280 = 1;
+    func_8010E9F0(4, 0);
+    gSaveContext.weekEventReg[0x8] |= 0x01;
+    func_80112AFC(globalCtx);
+    player->stateFlags1 |= 0x20;
+    this->actionFunc = EnMa4_SetupHorsebackGameWait;
+}
+
+void EnMa4_SetupHorsebackGameWait(EnMa4* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (globalCtx->interfaceCtx.unk_280 == 8) {
+        this->actionFunc = EnMa4_HorsebackGameWait;
+        player->stateFlags1 &= ~0x20;
+    }
+}
+
+void EnMa4_HorsebackGameWait(EnMa4* this, GlobalContext* globalCtx) {
+    static s16 D_80AC0258 = 0;
```

Comment:
```suggestion
    static s16 interfacePoppedBalloonCounter = 0;
```

Reply:
Why the `interface` part?
I was thinking in maybe dropping the `interface` part and prefix it with an `s` (because it's static)
```suggestion
    static s16 sPoppedBalloonCounter = 0;
```
What do you think?